pulsar-metadata/src/main/java/org/apache/pulsar/metadata/impl/PulsarZooKeeperClient.java

```diff
@@ -121,7 +126,11 @@ public ZooKeeper call() throws KeeperException, InterruptedException {
                         try {
                             newZk = createZooKeeper();
                         } catch (IOException ie) {
-                            log.error("Failed to create zookeeper instance to " + connectString, ie);
+                            log.error("Failed to create zookeeper instance to {}", connectString, ie);
+                            throw KeeperException.create(KeeperException.Code.CONNECTIONLOSS);
+                        } catch (QuorumPeerConfig.ConfigException e) {
+                            log.error("Failed to create zookeeper instance to {} with config path {}",
+                                    connectString, configPath, e);
```

Comment:
It can help to print out the config path regardless of the exception type. Something like:

```suggestion
                        } catch (IOException | QuorumPeerConfig.ConfigException e) {
                            log.error("Failed to create zookeeper instance to {} with config path {}",
                                    connectString, configPath, e);
```

Note that some changes are needed to compile it, as GitHub won't allow suggestions on deleted lines.

Reply:
Good point, addressed.